src/renderer/components/message/MessageListAndComposer.tsx

```diff
@@ -31,30 +31,54 @@ export default function MessageListAndComposer({
   )
 
   const onDrop = (e: React.DragEvent<any>) => {
-    const files = (e.target as any).files || e.dataTransfer.files
     e.preventDefault()
     e.stopPropagation()
-    const tx = window.translate
-
-    // TODO maybe add a clause here for windows because that uses backslash instead of slash
-    const forbiddenPathRegEx = /DeltaChat\/.+?\.sqlite-blobs\//gi
-    for (let i = 0; i < files.length; i++) {
-      const { path, name } = files[i]
-      if (!forbiddenPathRegEx.test(path.replace('\\', '/'))) {
-        openDialog('ConfirmationDialog', {
-          message: tx('ask_send_file_desktop', [name, chat.name]),
-          confirmLabel: tx('menu_send'),
-          cb: (yes: boolean) => {
-            if (!yes) {
-              return
-            }
-            DeltaBackend.call('messageList.sendMessage', chat.id, null, path)
-          },
-        })
-      } else {
-        log.warn('Prevented a file from being send again while dragging it out')
+    let sanitizedFileList: Pick<File, 'name' | 'path'>[] = []
+    {
+      let fileList: FileList = (e.target as any).files || e.dataTransfer.files
+      // TODO maybe add a clause here for windows because that uses backslash instead of slash
+      const forbiddenPathRegEx = /DeltaChat\/.+?\.sqlite-blobs\//gi
+      for (let i = 0; i < fileList.length; i++) {
+        const { path, name } = fileList[i]
+        // TODO filter out folders somehow
+        // if that is possible without a backend call to check wheter the file exists,
+        // maybe some browser api like FileReader could help
+        if (!forbiddenPathRegEx.test(path.replace('\\', '/'))) {
+          sanitizedFileList.push({ path, name })
+        } else {
+          log.warn(
+            'Prevented a file from being send again while dragging it out',
+            name
+          )
+        }
```

Comment:
I think for something like this functional patterns come in very handy.

Just some pseudo code

```
let fileList: FileList = (e.target as any).files || e.dataTransfer.files
const forbiddenPathRegEx = /DeltaChat\/.+?\.sqlite-blobs\//gi
const isForbidden = ({path}) => !forbiddenPathRegEx.test(path.replace('\\', '/'))
const sanitizedFileList = fileList.filter(isForbidden)

if (sanitizedFileList.length < fileList.length) {
  log.warn('Prevented some files from being sent')
}
```

Reply:
nah, we can still change this later.. 
PS: I used a feature of `let` here, scoped usage. might have interesting usages.